internal/codegen/tpl_stencil_arg.go

```diff
@@ -0,0 +1,222 @@
+// Copyright 2022 Outreach Corporation. All Rights Reserved.
+
+// Description: This file contains the code for the stencil.Arg
+// template function.
+
+package codegen
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/getoutreach/stencil/internal/dotnotation"
+	"github.com/getoutreach/stencil/pkg/configuration"
+	"github.com/pkg/errors"
+	"github.com/santhosh-tekuri/jsonschema/v5"
+)
+
+// Arg returns the value of an argument in the service's manifest
+//
+//	{{- stencil.Arg "name" }}
+func (s *TplStencil) Arg(pth string) (interface{}, error) {
+	if pth == "" {
+		// Deprecated: Arg(pth) will be required in the future. See note on
```

Comment:
Should probably track depreciations somewhere other than in a comment - they're bound to be forgotten about

Reply:
Agreed, I'll move it to the docs